transport/http_util.go

```diff
@@ -159,6 +161,8 @@ func newHPACKDecoder() *hpackDecoder {
 			}
 		case ":path":
 			d.state.method = f.Value
+		case "content-type":
```

Comment:
yeah, I don't think this is so compelling, and auto-switching probably doesn't belong here anyway. Let's expect users of non-proto encodings to specify their codec explicitly at a higher level.